### PR TITLE
fix(send): fill the full balance on Max instead of stranding a digit of dust

### LINKED ISSUE #4836: [Send] Coin.getMaxValue truncates to decimals−1 — token Max never fills the full balance
## Problem

`Coin.getMaxValue` truncates the computed max to `decimals − 1` fraction digits, so token Max never fills the full balance — one digit of dust is always left behind.

[Coin.swift#L322-L329](https://github.com/vultisig/vultisig-ios/blob/4d020d190/VultisigApp/VultisigApp/Model/Coin.swift#L322-L329):

```swift
return maxValueCalculated < .zero ? 0 : maxValueCalculated.truncated(toPlaces: decimals - 1)
```

Reached for token Max via `SendCryptoLogic.computeMaxAmount` → `SendDetailsViewModel.setMaxAmount`. For a token, `fee == 0`, so Max = balance truncated one digit short.

**Scenario:** USDT (6 dp) balance `100.123456` → Max fills `100.12345`, leaving `0.000006` USDT. Negligible value, but the Max chip can never zero a token balance.

## Suggested fix

The `decimals − 1` shave looks like an anti-overshoot guard, but the subtraction `rawBalance − fee` is exact BigInt — truncating to full `decimals` cannot overshoot. Return full precision (at minimum for non-native tokens where `fee == 0`).

Context: cross-platform "perfect max on all assets" sweep. The iOS send flow is otherwise correct (token validation is strict `>` with no fee mixed in; native max is re-derived on the Verify screen against fresh balance+fee).

### PR DESCRIPTION
**Pull request by gastonm5**
Fixes #4836

## Problem

`Coin.getMaxValue` truncated the computed max to `decimals - 1` places:

```swift
return maxValueCalculated < .zero ? 0 : maxValueCalculated.truncated(toPlaces: decimals - 1)
```

so every Max send left one unit-of-last-place behind. Non-native tokens never reach the native fee-refine path (`SendDetailsViewModel` guards it on `coin.isNativeToken`), which makes a token's Max exactly `balanceDecimal.truncated(toPlaces: decimals - 1)` — USDT (6 dp) with `100.123456` filled `100.12345`, and the dust was unreachable by any other means.

The shaved digit reads like an anti-overshoot guard but isn't one: `rawBalance - fee` is exact `BigInt` arithmetic and cannot overshoot, and the native Max is independently re-derived against a fresh balance + fee on the Verify screen.

## Change

Full precision for **all** coins — one rule, no native/token branch. Three sites carried the same defect:

- **`Model/Coin.swift`** — `getMaxValue` truncates to the coin's full `decimals`.
- **`Features/Send/Logic/SendCryptoLogic.swift`** — Terra Classic bypasses `getMaxValue` entirely (`computeMaxAmount` routes to `terraClassicMaxValue` for the burn-tax fixed point) and repeated the identical `coin.decimals - 1` shave. Same treatment: the fixed-point solve already funds the tax, so there is nothing left to pad against.
- **`Blockchain/Cardano/Signing/Cardano.swift`** — three validation-message sites hardcoded `toPlaces: 5` / `digits: 5` with comments explicitly mirroring the old rule ("ADA has 6 decimals, so decimals - 1 = 5"). They now quote ADA's full 6 dp through a named constant, so a suggested "Send Max" never advertises less than Max actually sends. The now-false comments are corrected.

Plus a stale comment fix in `SendValidationTests`.

## Broadcast path is safe

`Coin.raw(for:)` rounds **up** (`NSDecimalRound(&result, &decimal, 0, .up)`) and can exceed `rawBalance` by one unit at full precision — the latent issue tracked in #4835. **The Send flow never reaches it.** Every `raw(for:)` call site is in the Swap flow (`SwapCryptoLogic`, `SwapPayloadBuilder`, `SwapService`, the swap branches of `BlockChainService`, and the swap inbound-*fee* display in `SendSummaryViewModel`).

Send goes `tx.amountInRaw` → `SendCryptoLogic.amountInRaw` → `amountDecimal(...).truncated(toPlaces: coin.decimals)` → `.description.toBigInt(decimals:)`, and `String.toBigInt(decimals:)` truncates `.down` again. Every step rounds down, so full-precision amounts cannot produce an unbroadcastable transaction here.

## Note for reviewers: no visible ETH change

`SendCryptoLogic.computeMaxAmount` caps display formatting at `min(8, decimals)`, so chains with more than 8 decimals (ETH at 18) are **unchanged** by this fix — don't expect a visible difference there. The observable improvement is on ≤ 8-dp assets. `testComputeMaxAmountCapsDisplayAtEightDecimals` pins that cap as intentional, and `testGetMaxValueKeepsPrecisionBeyondDisplayCap` covers the full precision the cap hides at the `Coin` level.

## Gate

`make test` on iPhone 16 Pro Max — `** TEST SUCCEEDED **`, **3124 passed / 0 failed**.

No existing expectation needed changing: every assertion in `SendMaxAmountTests`, `SendValidationTests`, and `SendDetailsViewModelMaxAmountTests` was already exact at ≤ `decimals - 1` places.

New tests (7, in `SendMaxAmountTests`):

| Test | Covers |
|---|---|
| `testTokenMaxFillsFullBalance` | USDT 6 dp — `100.123456` fills all six |
| `testCardanoMaxFillsFullBalance` | ADA 6 dp |
| `testBitcoinMaxFillsFullBalanceToEightDecimals` | BTC 8 dp |
| `testNativeMaxWithFeeEqualsBalanceMinusFeeExactly` | native + nonzero fee, exact `balance − fee` |
| `testTerraClassicMaxFillsFullPrecisionAfterBurnTax` | `terraClassicMaxValue` fixed point |
| `testComputeMaxAmountCapsDisplayAtEightDecimals` | pins the `min(8, decimals)` display cap |
| `testGetMaxValueKeepsPrecisionBeyondDisplayCap` | 18-dp precision at the `Coin` level |

## Follow-up observed (not fixed here)

The Verify screen re-derives native Max as `balance − fee` without re-solving the Terra Classic fixed point, while `fee` embeds a burn tax computed from the *previous* amount. That can leave the tx short by 1 uluna. It is pre-existing and **this change improves it**: the deficit scales with the truncation residue (`candidate − A = 1.005·(A* − A)`), so finer truncation shrinks it. Simulated over 4000 balances, underfunded cases drop from 200/4000 (5 dp) to 20/4000 (6 dp), worst case unchanged at 1 uluna. Worth a separate issue.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated MAX-send amount calculations to use full coin precision (instead of leaving unit-of-last-place dust).
  * Cardano “Send Max” validation/recommendation messaging now quotes the fee-adjusted amount (balance minus estimated fee) with full 6-decimal ADA precision.
  * Terra Classic max-send math now preserves full precision.
* **Tests**
  * Added/expanded XCTest coverage for full-precision MAX-send behavior across multiple chains.
  * Added Cardano-specific tests to ensure messages use net-of-fee values and maintain 6-decimal formatting, including edge cases.

### CODE REVIEW

File: VultisigApp/VultisigApp/Blockchain/Cardano/Signing/Cardano.swift [hunk 1]
```diff
@@ -98,10 +104,10 @@ class CardanoHelper {
         // 2. Check sufficient balance
         let totalNeeded = sendAmount + estimatedFee
         if totalBalance < totalNeeded {
-            // For MAX amount display, truncate to 5 decimal places to match getMaxValue behavior
+            // Quote the suggested MAX at ADA's full precision, matching getMaxValue.
             let totalBalanceDecimal = totalBalance.toADA
-            let truncatedBalance = totalBalanceDecimal.truncated(toPlaces: 5) // ADA has 6 decimals, so decimals - 1 = 5
-            let totalBalanceADA = truncatedBalance.formatToDecimal(digits: 5)
+            let truncatedBalance = totalBalanceDecimal.truncated(toPlaces: adaDisplayDecimals)
+            let totalBalanceADA = truncatedBalance.formatToDecimal(digits: adaDisplayDecimals)
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Cardano/Signing/Cardano.swift]: Good catch — valid, and fixed in d346d4988.

Verified against `origin/main` first: `totalBalance.toADA` with no fee subtraction is unchanged there, so this is **pre-existing on main**, not introduced by this PR. It surfaced here because the branch already touches these exact lines to widen ADA precision from 5 to 6 decimals. Fixing it rather than deferring, since the whole point of this PR is that Max should be accurate — quoting an unachievable Max in the adjacent error message is the same class of wrongness.

Rather than repeat the subtraction at all three sites, extracted a helper:

```swift
private static func sendMaxSuggestionADA(totalBalance: BigInt, estimatedFee: BigInt) -> String {
    let maxLovelaces = Swift.max(BigInt(0), totalBalance - estimatedFee)
    let truncated = maxLovelaces.toADA.truncated(toPlaces: adaDisplayDecimals)
    return truncated.formatToDecimal(digits: adaDisplayDecimals)
}
```

All three call sites already sit behind guards that ensure `totalBalance > estimatedFee`, so the clamp is belt-and-braces rather than load-bearing.

New `CardanoSendMaxSuggestionTests` covers all three paths (insufficient balance, too-little-change, low-balance recommendation), the full-6-decimal case, and the fee-exceeds-balance guard. One note on how they are written: expectations are built through the same `formatToDecimal` the production code uses, because it takes its separator from `Locale.current` — asserting on a literal `"9.5"` would fail on a comma-decimal simulator.

Gate: `** TEST SUCCEEDED **`, 3130 executed / 1 skipped / 0 failures, all 5 new tests passing by name.

**Comment by gastonm5** [comment]
Follow-up filed: #4854 — the residual Terra Classic case. This PR improves it 10x (200/4000 → 20/4000 underfunded balances in simulation, worst case unchanged at 1 uluna) because the deficit scales with the truncation residue, but it does not eliminate it. The root cause is the Verify re-derivation not re-solving the burn-tax fixed point, which is pre-existing and out of scope here.

**Review by johnnyluo — APPROVED** [review]
LGTM
